From 56a9bd47f3f61b526400243c6f12f0e2401c5968 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Thu, 28 Mar 2013 15:05:33 -0700
Subject: [PATCH 253/974] MIPS: Add mips_cpunum() function.

This returns the CPUNum from the low order Ebase bits.

Signed-off-by: David Daney <david.daney@cavium.com>
[Original patch taken from Cavium SDK 3.1.2-568]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/mips/include/asm/mipsregs.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/mips/include/asm/mipsregs.h b/arch/mips/include/asm/mipsregs.h
index fb35995..d9b32cb 100644
--- a/arch/mips/include/asm/mipsregs.h
+++ b/arch/mips/include/asm/mipsregs.h
@@ -2160,6 +2160,11 @@ static inline unsigned long hypcall3(unsigned long num, unsigned long arg0,
 	return r;
 }
 
+static inline unsigned int mips_cpunum(void)
+{
+	return read_c0_ebase() & 0x3ff; /* Low 10 bits of ebase. */
+}
+
 #endif /* !__ASSEMBLY__ */
 
 #endif /* _ASM_MIPSREGS_H */
-- 
2.6.2

